lib/OpenQA/WebAPI/Auth/OpenID.pm

```diff
@@ -108,18 +108,23 @@ sub auth_response ($c) {
     };
 
     $csr->handle_server_response(
-        not_openid =>
-          sub () { $err_handler->('Failed to login', 'OpenID provider returned invalid data. Please retry again') },
```

Comment:
Just add the detail which openid server we connected, e.g.

```patch
- 'OpenID provider returned invalid data. Please retry again'
+ "OpenID provider '$url' returned invalid data. Please retry again"
```

and revert the rest

Reply:
Done